### ISSUE TITLE
RF05 warns for space in INSERT value

### ISSUE THREAD

**Issue by brainard52** [sqlfluff/sqlfluff#6614]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

Note: When searching for similar issues, I used `is:issue RF05 INSERT`.

### What Happened

I'm attempting to insert a value into a table, and RF05 is flagging for the value.

### Expected Behaviour

I expect data inserted for a TEXT column which contains spaces to be valid. 

I am very new to SQL and SQLite, so please let me know if this is expected behavior. The way I understood the documentation is that only identifiers such as table names were subject to this rule.

### Observed Behaviour

Text inserted into a table is flagged by RF05

### How to reproduce

Offending code:
```sql
CREATE TABLE "Foo" (
    id INTEGER PRIMARY KEY,
    name TEXT
);

INSERT INTO "Foo" (name) VALUES ("Bar Baz");
```

Output
```sh
$ sqlfluff lint --dialect sqlite files/sql/example.sql 
== [files/sql/example.sql] FAIL                                                                                         
L:   6 | P:  34 | RF05 | Do not use special characters in identifiers.
                       | [references.special_chars]
All Finished 📜 🎉!
```

### Dialect

`--dialect sqlite`

### Version

```sh
$ sqlfluff --version
sqlfluff, version 3.3.0
```

### Configuration

I do not have any configuration files.

```sh
$ ls -1A
.git
.mypy_cache
db.sqlite
files
linters.md
mypy.ini
pre-commit
start_standalone
stubs
test_deploy
tox.ini
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by brainard52** [comment]
I just realized I said LT05 instead of RF05. I'm going to edit my issue for that and then search for existing issues again.

**Comment by brainard52** [comment]
I have just been informed by a mentor that double quotes are reserved for identifiers while single quotes are reserved for strings. This is expected behavior!
